### DISCUSSION TITLE
Outline of the public API

### DISCUSSION THREAD

**Discussion by paulobruno** [worlddynamics/WorldDynamics.jl#101] [Ideas]
I created this discussion to try to outline the essential public API of WorldDynamcs.jl.
Our initial goal was to allow the user to plot each figure.
Then, we would like to easily change any part of a scenario, like initializations, parameters, tables, etc.
From #76, it seems that the user should also be able to access the solution, variables, and system of each figure.

My question now is: **what else?**

**Comment by paulobruno** [comment]
This is the current feature list in my mind:
1. Plot every figure;
2. Get the historical run of every system;
3. Edit the parameters of every system;
4. Edit the intializations of every system;
5. Edit the tables of every system;
6. Edit the equations of every system;
7. Insert equations into systems;
8. Get the solution of every figure;
9. Get the system of every figure;
10. Get the variables of every figure.

**Reply by natema** [reply]
That's good for me.

**Comment by paulobruno** [comment]
As @natema suggested in #118, we should also have "aliases" to figures using their book numbering instead of the modules path.
For example, we should be able to call the same figure 4.69a using both `World3.Agriculture.fig_69a()` or `World3.fig_4-69a()`.
